File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.java
class Solution {
    public int missingMultiple(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            map.put(nums[i], 1);
        }

        int max = 0;
        for(int i=0; i<nums.length; i++){
            max = Math.max(nums[i], max);
        }

        for(int i=k; i<=max; i+=k){
            if(!map.containsKey(i)) return i;
        }

        return ((max/k) + 1) * k;
    }
}

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        unordered_set<int> map;
        for(int i=0; i<nums.size(); i++){
            map.insert(nums[i]);
        }

        int mx = 0;
        for(int i=0; i<nums.size(); i++){
            mx = max(nums[i], mx);
        }

        for(int i=k; i<=mx; i+=k){
            if(map.find(i) == map.end(i)) return i;
        }

        return ((mx/k) + 1) * k;
    }
};